src/MCStack.cpp

```diff
@@ -414,16 +414,23 @@ MCStack::isInDeadlock() const
     const MCTransition &nextTransitionForTid =
       this->getNextTransitionForThread(tid);
 
-    if (nextTransitionForTid.ensuresDeadlockIsImpossible())
+    if (this->getThreadDataForThread(tid).getExecutionDepth() >=
```

Comment:
A deadlock is still possible with threads which have reached their maximum execution depth. We should already be handling this (see the next comment).

In fact, stopping if _any_ thread hits its maximum execution depth will potentially miss many deadlocks when we use `-max-execution-depth` because some deadlocks may only occur with some (but not all) threads reaching their execution depth.

Reply:
I'm not sure I understand the point here.  Let's talk.